### ISSUE TITLE
Delete line can make syntax unbalanced

### ISSUE THREAD

**Issue by gunnarahlberg** [cursiveclojure/cursive#732]
Keyboard shortcut can make the code parenthesis unbalanced. 

Steps to reproduce
Type
```
(def a
  "a")
```
Go to line two. 
Hit Ctrl+y to delete line

Expected: 

```
(def a
  )
```
Actual: 

```
(def a
```

**Comment by cursiveclojure** [comment]
Yeah, this is a bit tricky. Should Cursive delete everything before the paren? Everything after it? Everything except the paren?

**Comment by cursiveclojure** [comment]
Thinking about it, I think what you really want is to delete everything on the line except the paren, and put the paren at the end of the previous line, so you'd get:
```
(def a)
```

**Comment by gunnarahlberg** [comment]
I'd say everything on the line but the paren



-------- Ursprungligt meddelande --------
Ämne: Re: [cursive] Delete line can make syntax unbalanced (#732)
Från: cursiveclojure <notifications@github.com>
Till: cursiveclojure/cursive <cursive@noreply.github.com>
Kopia: Gunnar Ahlberg <github@gunnarahlberg.com>

**Comment by gunnarahlberg** [comment]
Oh so elegant! 

-------- Ursprungligt meddelande --------
Ämne: Re: [cursive] Delete line can make syntax unbalanced (#732)
Från: cursiveclojure <notifications@github.com>
Till: cursiveclojure/cursive <cursive@noreply.github.com>
Kopia: Gunnar Ahlberg <github@gunnarahlberg.com>
